builder/json.go

```diff
@@ -0,0 +1,84 @@
+package builder
+
+import "encoding/json"
+
+type JSON struct {
+	Typ    ComponentType
+	Fields map[string]interface{}
```

Comment:
maybe let's name it spec ?

Reply:
Some options:
- `JSON`/`JSONComponent`
- back to `Generic`/`GenericComponent`
- `Spec`/`SpecComponent` (I don't quite like it)
- `Custom`/`CustomComponent`
- `NonStandardComponent`
- `Any`/`AnyComponent`

I like the `Custom` and the `AnyComponent`, WDYT?

Reply:
I'd name it spec because that's the Druid semantic https://github.com/search?q=repo%3Aapache%2Fdruid%20spec&type=code

Reply:
That makes sense. Done!